plugins/src/signaller/imp.rs

```diff
@@ -93,7 +65,10 @@ impl Signaller {
         let (mut ws_sink, mut ws_stream) = ws.split();
 
         ws_sink
-            .send(WsMessage::Text("REGISTER PRODUCER".to_string()))
+            .send(WsMessage::Text(
+                serde_json::to_string(&p::IncomingMessage::Register(p::RegisterMessage::Producer))
```

Comment:
Maybe somewhere at the very top you want to wrap a [`with(|m| serde_json::to_string(m).unwrap())`](https://docs.rs/futures/latest/futures/sink/trait.SinkExt.html#method.with) and here to not repeat yourself. Same for the stream